src/com/company/Models/Users/CourseAdministrator.java

```diff
@@ -148,10 +149,98 @@ public void renameCourseModule(ArrayList<CourseModule> courseModules, CourseModu
+        currentCourseModuleResultsAsArray.removeAll(Collections.singleton(null));
+
+        for (CourseModuleResult currentCourseModuleResult: currentCourseModuleResultsAsArray) {
+            CourseModule currentCourseModule =
+                    new CourseModuleLoader().loadCourseModule(currentCourseModuleResult.getCourseModuleCode());
+            if (currentCourseModule.getLevel() == student.getLevel()) {
+                courseModulesForLevel.add(currentCourseModuleResult);
+            }
+        }
+
+        ArrayList<CourseModuleResult> completedCourseModuleResults = student.getCompletedCourseModules();
+
+        for (CourseModuleResult completedCourseModuleResult: completedCourseModuleResults) {
+            CourseModule completedCourseModule =
+                    new CourseModuleLoader().loadCourseModule(completedCourseModuleResult.getCourseModuleCode());
+
+            if (completedCourseModule.getLevel() == student.getLevel()) {
+                courseModulesForLevel.add(completedCourseModuleResult);
+            }
+        }
+
+        for (CourseModuleResult courseModuleResult: courseModulesForLevel) {
+            CourseModule courseModule = new CourseModuleLoader().loadCourseModule(courseModuleResult.getCourseModuleCode());
+            currentLevelTable.addRow(null, courseModule.getName() + " (" + courseModuleResult.getCourseModuleCode() + ")");
+            currentLevelTable.addRule();
+
+            for (String assignmentId: courseModuleResult.getAssignmentResults().keySet()) {
+                Assignment assignment = new AssignmentLoader().loadAssignment(assignmentId);
+
+                currentLevelTable.addRow(
+                        assignment.getAssignmentName(),
+                        courseModuleResult.getAssignmentResults().get(assignmentId) + "/" + assignment.getTotalPossibleMarks()
+                );
+                currentLevelTable.addRule();
+            }
+
+            currentLevelTable.addRow(
+                    "",
+                    "Total: "
+                            + courseModuleResult.getTotalMark()
+                            + "/"
+                            + courseModule.getTotalAvailableMarks()
+            );
+            currentLevelTable.addRule();
+
+            String passOrFail =
+                    ((double) courseModuleResult.getTotalMark() / (double) courseModule.getTotalAvailableMarks() * 100) > 40 ? "Pass" : "Fail";
+            String status = completedCourseModuleResults.contains(courseModuleResult) ? passOrFail : "In Progress";
+            currentLevelTable.addRow(
+                    "",
+                    "Pass/Fail/In Progress: "
+                            + status
+            );
+            currentLevelTable.addRule();
+        }
+
+        resultsSlip.append(currentLevelTable.render());
+
+        //TODO: Can student progress onto the next level of study?
+        boolean canProgressToNextLevel = student.canProgressToNextLevel();
```

Comment:
Might as well utilise the canProgressToNextLevel method to let course administrators promote students, maybe separate task.